### PR TITLE
Encrypt ECDSA signatures in release builds

### PR DESCRIPTION
**Pull request by klochowicz**
First commit adds a CI job that demonstrate the failing tests in release build (with relevant feature flag enabled needed for the tests to run).

### CODE REVIEW

File: src/zkp/ecdsa_adaptor.rs [hunk 1]
```diff
@@ -198,17 +197,16 @@ impl EcdsaAdaptorSignature {
...
+            let res = ffi::secp256k1_ecdsa_adaptor_encrypt(
+                *secp.ctx(),
+                &mut adaptor_sig,
+                sk.as_c_ptr(),
+                enckey.as_c_ptr(),
+                msg.as_c_ptr(),
+                ffi::secp256k1_nonce_function_ecdsa_adaptor,
+                aux_rand.as_c_ptr() as *mut ffi::types::c_void,
             );
+            debug_assert_eq!(res, 1);
```
Comment (thomaseizinger) [inline src/zkp/ecdsa_adaptor.rs:209]: To be consistent with other files in this repo, can we have the `unsafe` block evaluate to `res`?

That will also keep the unsafe as short as possible.

**Review by thomaseizinger — APPROVED** [review]
ACK 37c6afcf3ead55cbc99cbbba459745c43c7197a3

**Review by thomaseizinger — APPROVED** [review]
ACK e8692101e357fbda0a613023e0cd3e36457d9bf4

**Review by jonasnick — COMMENTED** [review]
Good catch @klochowicz!

ACK e8692101e357fbda0a613023e0cd3e36457d9bf4

**Pull request merged**